tests/Gaufrette/Functional/Adapter/FunctionalTestCase.php

```diff
@@ -67,8 +67,6 @@ public function shouldWriteAndRead()
 
         $this->assertEquals('Some content', $this->filesystem->read('foo'));
         $this->assertEquals('Some content1', $this->filesystem->read('test/subdir/foo'));
-        $this->filesystem->delete('foo');
```

Comment:
Yep there are useful because it's not the place where such clean up tasks should happen: it makes tests flaky since any failure would mark test as failed, even if it's not. I've seen such behavior with Azure adapter. Also every maintained adapter has a dedicated cleanup procedure, and IMHO other tests should be removed.